### PR TITLE
Move MOM_generic_tracer out of MOM6

### PR DESCRIPTION
**Pull request by theresa-cordero**
Move MOM_generic_tracer out of MOM6 and into the externals as a stub. With this change, users of the generic tracer modules will be requied to add an active version of MOM_generic_tracer to their equivalent of the GFDL_ocean_BGC folder.

This should not impact users running without generic tracers.

This PR is done in two commits, one where the stub has comments and extra use statements which are removed in the second commit.

### CODE REVIEW

**Comment by theresa-cordero** [comment]
This PR is related to [PR 107](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/pull/107). 

If a version of MOM6 with this change is used with a version of ocean_BGC without the changes in PR 107, then the model will fail to compile.

File: config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90 [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+!use g_tracer_utils,   only: g_tracer_get_name,g_tracer_set_values,g_tracer_set_common,g_tracer_get_common
+!use g_tracer_utils,   only: g_tracer_get_next,g_tracer_type,g_tracer_is_prog,g_tracer_flux_init
+!use g_tracer_utils,   only: g_tracer_send_diag,g_tracer_get_values
+!use g_tracer_utils,   only: g_tracer_get_pointer,g_tracer_get_alias,g_tracer_set_csdiag
+!use g_tracer_utils,   only: g_tracer_get_obc_segment_props
+use g_tracer_utils,   only: g_tracer_type
+
+use MOM_ALE_sponge, only : ALE_sponge_CS
+use MOM_coms, only : EFP_type
```
Comment (yichengt900) [inline config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90:33]: Remove the trailing space at the end of the line.

File: config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90 [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+!use g_tracer_utils,   only: g_tracer_get_name,g_tracer_set_values,g_tracer_set_common,g_tracer_get_common
+!use g_tracer_utils,   only: g_tracer_get_next,g_tracer_type,g_tracer_is_prog,g_tracer_flux_init
+!use g_tracer_utils,   only: g_tracer_send_diag,g_tracer_get_values
+!use g_tracer_utils,   only: g_tracer_get_pointer,g_tracer_get_alias,g_tracer_set_csdiag
+!use g_tracer_utils,   only: g_tracer_get_obc_segment_props
+use g_tracer_utils,   only: g_tracer_type
+
+use MOM_ALE_sponge, only : ALE_sponge_CS
+use MOM_coms, only : EFP_type 
+use MOM_diag_mediator, only : diag_ctrl
```
Comment (yichengt900) [inline config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90:34]: Same here. a trailing space is detected at the end of the line

File: config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90 [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+use MOM_coms, only : EFP_type 
+use MOM_diag_mediator, only : diag_ctrl 
+use MOM_file_parser, only : param_file_type
+use MOM_forcing_type, only : forcing, optics_type
+use MOM_grid, only : ocean_grid_type
+use MOM_hor_index, only : hor_index_type
+use MOM_open_boundary, only : ocean_OBC_type
+use MOM_restart, only : MOM_restart_CS
+use MOM_sponge, only : sponge_CS
+use MOM_time_manager, only : time_type
```
Comment (yichengt900) [inline config_src/external/GFDL_ocean_BGC/MOM_generic_tracer.F90:42]: Same. Remove the trailing space at the end of the line

**Comment by yichengt900** [comment]
@theresa-morrison, as you suggested, we can start with our first PR by simply moving `MOM_generic_tracer` into `external`. I have created a new branch, `dev/cefi_to_gfdl`, and you can resubmit the PR to that branch.

**Comment by theresa-cordero** [comment]
Closing this PR for now, a new PR has been opened to the new branch.

**Pull request closed without merging**